python_lib/qss_compiler/__init__.py

```diff
@@ -37,3 +32,9 @@
     link_file,
     LinkOptions,
 )
+
+from .exceptions import (  # noqa: F401
+    QSSCompilationFailure,
+    QSSCompilerError,
+    QSSLinkingFailure,
+)
```

Comment:
I remember Thomas not wanting these to be globally available in the top level init (https://github.com/openqasm/qe-compiler/pull/124#discussion_r1254601580). We left `QSSCompilationFailure` and `QSSCompilerError` for backward compatibility.